### PR TITLE
Feature/hardware wallet interface

### PR DESCRIPTION
**Pull request by piyalbasu**
[ticket](https://stellarorg.atlassian.net/browse/WAL-1223?atlOrigin=eyJpIjoiNzQyZTQ1MzE5ZjlkNDVkMmE2ZWExMDk1ODE1ODhjYjEiLCJwIjoiaiJ9)

This PR creates an interface so hardware wallets in the ecosystem can quickly add themselves to Freighter by adding their wallet details to a couple of different configuration objects. The Ledger integration we previously had now becomes an example of how to use the interface

### CODE REVIEW

File: extension/INTEGRATING_HARDWARE_WALLET.MD [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+   YOUR_WALLET = "Your Wallet",
+   NONE = "",
+   }
+   ```
+
+2. Navigate to
+   [src/popup/helpers/hardwareConnect.ts](src/popup/helpers/hardwareConnect.ts).
+   Here we'll add configurations for the new wallet.
+
+3. First we'll add helper methods so Freighter knows how to interact the
```
Comment (aristidesstaffieri) [inline extension/INTEGRATING_HARDWARE_WALLET.MD:32]: small typo, `knows how to interact the` -> `knows how to interact with the`

File: extension/INTEGRATING_HARDWARE_WALLET.MD [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+   the transaction using prompts on their hardware wallet. Once complete, the
+   API will return a signature to Freighter. Freighter will then add that
+   signature to the transaction.
+
+6. With those methods set, Freighter now knows how to interact with the hardware
+   wallet. Now, we'll tell Freighter how to show information about your wallet.
+   Scroll down to `walletAssets`. Here we'll set the key `logo` with your
+   hardware wallet's logo. This will appear in a button when Freighter lists all
+   possible hardware wallets to connect with. Drop your assets into the
+   [src/popup/assets/](src/popup/assets/) folder. You can use almost any image
```
Comment (aristidesstaffieri) [inline extension/INTEGRATING_HARDWARE_WALLET.MD:73]: Should we document or enforce specific dimensions or resolutions in order to create some consistency across assets from different providers?

File: extension/src/popup/views/AddAccount/connect/DeviceConnect/index.tsx [hunk 3]
```diff
@@ -104,11 +105,7 @@ export const DeviceConnect = () => {
         <div className="DeviceConnect__content__center">
           <img
             className="DeviceConnect__img"
-            src={
-              walletConnectSuccessful
-                ? walletAssets[walletType].connected
-                : walletAssets[walletType].idle
-            }
+            src={walletConnectSuccessful ? LedgerConnected : Ledger}
```
Comment (aristidesstaffieri) [inline extension/src/popup/views/AddAccount/connect/DeviceConnect/index.tsx:108]: Same as the other one, should this be keyed by active hardware wallet type?

File: extension/src/popup/ducks/transactionSubmission.ts [hunk 4]
```diff
@@ -164,35 +164,32 @@ export const submitFreighterSorobanTransaction = createAsyncThunk<
...
       const tx = TransactionBuilder.fromXDR(transactionXDR, networkPassphrase);
 
-      const transport = await TransportWebUSB.create();
-      const ledgerApi = new LedgerApi(transport);
-      const result = await ledgerApi.signTransaction(
-        bipPath,
-        tx.signatureBase(),
-      );
+      const signature = await hardwareSign[walletType]({ bipPath, tx });
+      console.log(signature);
```
Comment (aristidesstaffieri) [inline extension/src/popup/ducks/transactionSubmission.ts:187]: leftover log?

**Review by aristidesstaffieri — APPROVED** [review]
left a few questions but lgtm to me otherwise

File: extension/INTEGRATING_HARDWARE_WALLET.MD (hunk 2 above)
Comment (piyalbasu) [reply extension/INTEGRATING_HARDWARE_WALLET.MD:74]: We do, actually. I list the recommended dimensions in the helper file

File: extension/src/popup/components/hardwareConnect/HardwareSign/index.tsx [hunk 5]
```diff
@@ -138,11 +139,7 @@ export const HardwareSign = ({
           <div className="HardwareSign__content__center">
             <img
               className="HardwareSign__img"
-              src={
-                hardwareConnectSuccessful
-                  ? walletAssets[walletType].signing
-                  : walletAssets[walletType].idle
-              }
+              src={hardwareConnectSuccessful ? LedgerSigning : Ledger}
```
[the remark this answers is not in the document]
Comment (piyalbasu) [reply extension/src/popup/components/hardwareConnect/HardwareSign/index.tsx:142]: I'm opening a design ticket to address this

File: extension/INTEGRATING_HARDWARE_WALLET.MD (hunk 2 above)
Comment (aristidesstaffieri) [reply extension/INTEGRATING_HARDWARE_WALLET.MD:74]: oh I missed that, thanks.

**Pull request merged**